keras_cv/models/object_detection/retinanet/retinanet_test.py

```diff
@@ -244,6 +244,49 @@ def test_saved_model(self, save_format, filename):
         restored_output = restored_model(input_batch)
         self.assertAllClose(model_output, restored_output)
 
+    @pytest.mark.large
+    def test_custom_saved_model(self, save_format, filename):
+        class CustomPredictionHead(
```

Comment:
Could we make sense simple convolutions or something with real variables?

Reply:
Also, how does the forward pass work if these classes are ill-defined?

Reply:
I ran the test and it worked, because the custom classes are inheriting from the classes that are used as default in the RetinaNet class, they just have different names. If you think we should also test for custom architecture, e.g. having separable convolutions instead of regular ones, I'll do it.

Reply:
We could also test for custom Label Encoder, Label Decoder, Anchor Generator, Backbone, and so on.